docs/src/2.3.RenderingObjects.mdx

```diff
@@ -0,0 +1,214 @@
+      scale: { x: idx + 1, y: idx + 1, z: idx + 1 },
+      // set cube to a random color
+      color: { r: Math.random(), g: Math.random(), b: Math.random(), a: 1 },
+    };
+  });
+
+  return (
+    <Worldview>
+      <button style={{ position: "absolute", top: 0, left: 0 }} onClick={() => setCount(count + 1)}>
+        Add a Cube
+      </button>
+      <Cubes>{sphereMarkers}</Cubes>
+      <Axes />
+    </Worldview>
+  );
+}
+```
+
+Whenever you click the "Add a Cube" button, you'll see a new cube with a randomly generated color added to the scene.
+
+<CodeSandboxEmbed src="https://codesandbox.io/s/lxzrxqlpym" />
+
+## Rendering a Colorful Knot
+
+Now let's render something more interesting!
+
+A [trefoil knot](https://en.wikipedia.org/wiki/Trefoil_knot) is like a common overhand knot with its two loose ends joined, resulting in a knotted loop. In this step, we'll use parametric equations to render a trefoil knot with some color variation.
+
+```js
+// src/Example.js
+function Example() {
+  const steps = 500; // total number of objects
+
+  // map a number/index to a specific color
+  function numberToColor(number, max, a = 1) {
+    const i = (number * 255) / max;
+    const r = Math.round(Math.sin(0.024 * i + 0) * 127 + 128) / 255;
+    const g = Math.round(Math.sin(0.024 * i + 2) * 127 + 128) / 255;
+    const b = Math.round(Math.sin(0.024 * i + 4) * 127 + 128) / 255;
+    return { r, g, b, a };
+  }
+
+  // the object index needs to be multipled by this scale so it's evenly distributed in the space
+  const scale = (Math.PI * 2) / steps;
+  const sphereMarkers = new Array(steps)
+    .fill()
+    .map((_, idx) => [
+      // generate x, y, z coordinates based on trefoil equation
+      Math.sin(idx * scale) + 2 * Math.sin(2 * idx * scale),
+      Math.cos(idx * scale) - 2 * Math.cos(2 * idx * scale),
+      -Math.sin(3 * idx * scale),
+    ])
+    .map(([x, y, z], idx) => ({
+      pose: {
+        orientation: { x: 0, y: 0, z: 0, w: 1 },
+        position: { x: 20 * x, y: 20 * y, z: 20 * z },
+      },
+      scale: { x: 1, y: 1, z: 1 },
+      color: numberToColor(idx, steps),
+    }));
+
+  return (
+    <Worldview>
+      <Spheres>{sphereMarkers}</Spheres>
+      <Axes />
+    </Worldview>
+  );
+}
+```
+
+<CodeSandboxEmbed src="https://codesandbox.io/s/2x26ov87j0" />
+
+## Improving Performance with Instanced Rendering
+
+When drawing many instances of the same object, you'll quickly reach a performance bottleneck with the number of GPU calls. [Instancing](https://learnopengl.com/Advanced-OpenGL/Instancing) allows us to draw many objects at once with a single render call, saving us valuable CPU–GPU communication time.
+
+Most of the existing [Commands](#/docs/api/command) support instancing, which is why they accept **arrays** of objects as children. In this step, let's convert our previous example to use instanced rendering.
+
+```js
+// src/Example.js
+function Example() {
+  // ... other existing code
+  new Array(steps)
```

Comment:
I was confused about where `sphereMarker` came from, maybe change that to `const sphereMarker = [];` since it's not returned from the array any more?